flink-tests/src/test/java/org/apache/flink/test/scheduling/SpeculativeExecutionITCase.java

```diff
@@ -524,30 +542,66 @@ public void open(GenericInputSplit split) throws IOException {
     }
 
     private static class TestingNumberSequenceSource extends NumberSequenceSource {
-        private TestingNumberSequenceSource() {
+
+        private final boolean forceFailureFlag;
```

Comment:
Looks to me it can be replaced by `forceFailureCounter`(forceFailureCounter=0 means forceFailureFlag=false).

Reply:
Maybe add some comments to explain what will happen if `forceFailureCounter > 0`?